core/src/test/java/org/mskcc/cbio/portal/util/TestRequestBodyGZipFilter.java

```diff
@@ -0,0 +1,174 @@
+        ArgumentCaptor<HttpServletResponse> responseArg = ArgumentCaptor.forClass(HttpServletResponse.class);
+        Mockito.verify(chain, Mockito.times(1)).doFilter(requestArg.capture(), responseArg.capture());
+        RequestBodyGZipFilter.GzippedInputStreamWrapper wrappedStream = requestArg.getValue();
+        
+        String actualRequestBody = wrappedStream.getReader().lines().collect(Collectors.joining());
+        String expectedRequestBody = "{\"field\":\"value\"}";
+        Assert.assertEquals(expectedRequestBody, actualRequestBody);
+    }
+
+    /**
+     * When I filter a request that has the gzip header and the gzip
+     * feature is enabled and the method is POST, but the unzipped message
+     * body is larger than maxInflatedRequestBodySize, the doFilter method
+     * should wrap the request with a GzippedInputStreamWrapper.
+     * The BufferedReader for this GzippedInputStreamWrapper should throw an
+     * IOException as it is read. This exception gets turned into
+     * an UncheckedIOException by something, so we'll check for that instead.
+     */
+    @Test(expected = UncheckedIOException.class)
+    public void testDoFilterEnabledGZippedButTooLong() throws Exception {
+        RequestBodyGZipFilter subject = createRequestBodyGZipFilter(true, 1);
+
+        FileInputStream inputStream = new FileInputStream("src/test/resources/gzip_filter.json.gz");
+        TestInputStream stream = new TestInputStream(inputStream);
+        HttpServletRequest request = createHttpServletRequest("gzip", "POST");
+        Mockito.when(request.getInputStream()).thenReturn(stream);
+
+        HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+        FilterChain chain = Mockito.mock(FilterChain.class);
+
+
+        subject.doFilter(request, response, chain);
+
+        // Request should be wrapped, so the request should now be a GzippedInputStreamWrapper
+        ArgumentCaptor<RequestBodyGZipFilter.GzippedInputStreamWrapper> requestArg =
+            ArgumentCaptor.forClass(RequestBodyGZipFilter.GzippedInputStreamWrapper.class);
+        ArgumentCaptor<HttpServletResponse> responseArg = ArgumentCaptor.forClass(HttpServletResponse.class);
+        Mockito.verify(chain, Mockito.times(1)).doFilter(requestArg.capture(), responseArg.capture());
+        RequestBodyGZipFilter.GzippedInputStreamWrapper wrappedStream = requestArg.getValue();
+
+        //this should cause an IOException
+        wrappedStream.getReader().lines().collect(Collectors.joining());
+    }
+
+
+    private RequestBodyGZipFilter createRequestBodyGZipFilter(boolean enabled, int requestBodySize) {
+        RequestBodyGZipFilter subject = new RequestBodyGZipFilter();
+        ReflectionTestUtils.setField(subject, "maxInflatedRequestBodySize", requestBodySize);
+        ReflectionTestUtils.setField(subject, "enabled", enabled);
+        return subject;
+    }
+    
+    private HttpServletRequest createHttpServletRequest(String gzip, String post) {
```

Comment:
the "post" argument name is improper ... sometimes the string coming in is "GET", so rename to "command" or "httpCommand"

Reply:
This is what happens when you click through Intellij's method refactoring too quickly. I've changed it to `httpMethod`